rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -2624,6 +2624,21 @@ bool spu_thread::do_putllc(const spu_mfc_cmd& args)
 		const auto& to_write = _ref<spu_rdata_t>(args.lsa & 0x3ff80);
 		auto& res = vm::reservation_acquire(addr);
 
+		if (cmp_rdata(to_write, rdata))
```

Comment:
This should be under the rsx lock below. Some games are not properly synchronized, which is why I have the "limit scope" warning below. I could not use a smaller scope without side effects.

Reply:
Or maybe we can ask users to verify, the code has changed too much since it was originally written, no idea if its still an issue. Needs a fairly slow CPU with no TSX though.